cpp/open3d/core/CUDAUtils.cpp

```diff
@@ -96,6 +131,7 @@ int GetDevice() {
 }
 
 static void SetDevice(int device_id) {
+    AssertCUDADeviceAvailable(device_id);
```

Comment:
this line is the minimum change, that triggers this behavior.

Reply:
Great, we can dig into `AssertCUDADeviceAvailable` to see which line causes the issue.